### PR TITLE
NEXUS-4570, adding catch for FileNotFoundException

### PR DESCRIPTION
**Pull request by bdemers**
No real way to write a test for this

### CODE REVIEW

**Comment by peterlynch** [comment]
this usually means code is written to be untestable. You could refactor the code or use mocks and make sure ItemNotFoundException is thrown when your mock throws FileNotFoundException.

These are subtle changes that are affecting what can get thrown from this method. I would try to write a test for it. At the very least, add a simple comment next to code referencing what you are fixing.

File: nexus/nexus-proxy/src/main/java/org/sonatype/nexus/proxy/storage/local/fs/DefaultFSLocalRepositoryStorage.java [hunk 1]
```diff
@@ -325,6 +326,10 @@ public class DefaultFSLocalRepositoryStorage
                         request, file );
                 }
             }
+            catch ( FileNotFoundException e )
```
Comment (velo) [inline nexus/nexus-proxy/src/main/java/org/sonatype/nexus/proxy/storage/local/fs/DefaultFSLocalRepositoryStorage.java:329]: Looking at code this looks like an unreachable branch.

This is probably a multi thread situation where on thread is running other went and deleted the file.  Best case scenario we need a TODO to have  better locking on files.

Comment (bdemers) [inline nexus/nexus-proxy/src/main/java/org/sonatype/nexus/proxy/storage/local/fs/DefaultFSLocalRepositoryStorage.java:329]: I can add a better comment.  And it's _possible_ the file is getting removed by another process

**Comment by velo** [comment]
+1

**Comment by peterlynch** [comment]
+1 nice changes

File: nexus/nexus-proxy/src/main/java/org/sonatype/nexus/proxy/storage/local/fs/DefaultFSLocalRepositoryStorage.java (hunk 1 above)
Comment (cstamas) [inline nexus/nexus-proxy/src/main/java/org/sonatype/nexus/proxy/storage/local/fs/DefaultFSLocalRepositoryStorage.java:329]: The file is definitely deleted by something else that is not using UID locking, and that's the culprit. This change here only makes the symptom not visible. But the ultimate question is what modifies the repo in RW manner and avoids locks?

**Pull request merged**